File: program/program/find-digital-root-of-a-number/FindDigitalRootOfANumber.c
// C Program to find the Digital Root of a Number
#include<stdio.h>
int digital_root(int);
void main()
{
    int num;
    printf("Enter Your Number :\n");
    scanf("%d",&num);
    if(num!=0){
    int dr=digital_root(num);
    printf("Digital Root of %d is %d",num,dr);
    }
}
int digital_root(int n)
{
    int sum=0;
    while(1)
    {
      while(n!=0)
      {
         sum=sum+(n%10);
         n=n/10;
      }
      if(sum>9)
      {
        n=sum;
        sum=0;
      }
      else
        break;
    } 
    return sum;
}

File: program/program/find-digital-root-of-a-number/find_digital_root_of_a_number.go
package main

import "fmt"

func findDigitalRoot(num int) int {
	temp := num
	digit := 0
	droot := 0
	for temp != 0 {
		digit = temp % 10
		droot += digit
		temp /= 10
		if temp == 0 && droot > 9 {
			temp = droot
			droot = 0
		}
	}
	fmt.Println(droot)
	return droot
}
func main() {
	fmt.Println("Enter the number to find its digital root: ")
	var num int
	fmt.Scanln(&num)
	findDigitalRoot(num)
}
